src/core/dbus-execute.c

```diff
@@ -910,7 +981,7 @@ int bus_set_transient_exec_command(
                 if (r < 0)
                         return r;
 
-                r = sd_bus_message_read(message, "b", &b);
+                r = !is_ex_prop ? sd_bus_message_read(message, "b", &b) : sd_bus_message_read_strv(message, &ex_opts);
```

Comment:
not that it matters, but maybe swap the two branches of the ternary argument. then you can drop the extra `!` and simplify things a tiny bit